ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -458,6 +479,52 @@ def yield_view_lineage(self) -> Optional[Iterable[AddLineageRequest]]:
                     f"Could not parse query [{view_definition}] ingesting lineage failed: {exc}"
                 )
 
+    def _get_foreign_constraints(
+        self, table_constraints: OMetaTableConstraints
+    ) -> List[TableConstraint]:
+        """
+        Search the referred table for foreign constraints
+        and get referred column fqn
+        """
+
+        foreign_constraints = []
+        for constraint in table_constraints.foreign_constraints:
+            referred_column_fqns = []
+            referred_table = fqn.search_table_from_es(
+                metadata=self.metadata,
+                table_name=constraint.get("referred_table"),
+                schema_name=constraint.get("referred_schema"),
+                database_name=None,
+                service_name=self.context.database_service.name.__root__,
+            )
+            for column in constraint.get("referred_columns"):
+                col_fqn = get_column_fqn(table_entity=referred_table, column=column)
+                if col_fqn:
+                    referred_column_fqns.append(col_fqn)
+            foreign_constraints.append(
+                TableConstraint(
+                    constraintType=ConstraintType.FOREIGN_KEY,
+                    columns=constraint.get("constrained_columns"),
+                    referredColumns=referred_column_fqns,
```

Comment:
if the referred columns are not found still we will create the table constraint without the reference!